### PR TITLE
MicrosoftPolicyAndComplianceAuditLog - Replace Deprecated Module

### PR DESCRIPTION
**Pull request by jlevypaloalto**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [MicrosoftPolicyAndComplianceAuditLog - Replace Deprecated Module](https://jira-hq.paloaltonetworks.local/browse/CIAC-6282)

## Description
This PR changes the O365 client from the older version of the Exchange Online PowerShell module which uses the depreciated RPS to the latest V3 module.

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [x] Yes
       - Further details: Instances of this integration will have to be reconfigured with the new parameters.
   - [ ] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Review by JudahSchwartz — CHANGES_REQUESTED** [review]
we decided that we're not doing a v2, can we change the pr to reflect that? This will make it easier for me to see what was changed

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/README.md
Comment (amshamah419) [inline Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/README.md:2]: Remove

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/README.md
Comment (amshamah419) [inline Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/README.md:108]: Let's leave the example since it will help users better understand.

**Review by amshamah419 — CHANGES_REQUESTED** [review]
Generally speaking, what was the decision regarding breaking changes?

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/README.md
Comment (jlevypaloalto) [reply Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/README.md:108]: Sure, i havent finished with the docs just yet, it was decided to stay on V1 and make it clear to the users that their instances will have to be reconfigured.

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/README.md
Comment (jlevypaloalto) [reply Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/README.md:2]: my bad, ill get rid of it.

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml
Comment (JudahSchwartz) [inline Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml:9]: why wasnt this required before? @amshamah419

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml
Comment (JudahSchwartz) [inline Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml:164]: why the docker switch?

Comment (amshamah419) [reply Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml:164]: This is expected with the library change.

File: Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml
Comment (amshamah419) [reply Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.yml:9]: ¯\_(ツ)_/¯

**Comment by jlevypaloalto** [comment]
A force-merge is needed as this PR has breaking changes.

**Review by JudahSchwartz — APPROVED** [review]
Great! please wait on @amshamah419's approval

File: Packs/Office365AndAzureAuditLog/ReleaseNotes/1_0_5.md
Comment (JudahSchwartz) [inline Packs/Office365AndAzureAuditLog/ReleaseNotes/1_0_5.md:4]: lets make it very clear that users will need to reconfigure the instance

File: Packs/Office365AndAzureAuditLog/pack_metadata.json
Comment (JudahSchwartz) [inline Packs/Office365AndAzureAuditLog/pack_metadata.json:5]: this needs a major bump

File: Packs/Office365AndAzureAuditLog/pack_metadata.json
Comment (JudahSchwartz) [inline Packs/Office365AndAzureAuditLog/pack_metadata.json:5]: 2.0.0

Comment (jlevypaloalto) [reply Packs/Office365AndAzureAuditLog/pack_metadata.json:5]: good point

File: Packs/Office365AndAzureAuditLog/ReleaseNotes/2_0_0.json
Comment (ShahafBenYakir) [inline Packs/Office365AndAzureAuditLog/ReleaseNotes/2_0_0.json:1]:
```suggestion
{"breakingChanges":true,"breakingChangesNotes":"All integration instances of Microsoft Policy And Compliance (Audit Log) must be re-configured with the new fields."}
```

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**
